### PR TITLE
ambient: test split service

### PR DESCRIPTION
**Pull request by howardjohn**
Fixes https://github.com/istio/istio/issues/43189

### CODE REVIEW

File: pilot/pkg/xds/endpoint_builder.go [hunk 1]
```diff
@@ -468,10 +458,44 @@ func buildEnvoyLbEndpoint(b *EndpointBuilder, e *model.IstioEndpoint) *endpoint.
 			},
 		}
 	}
+	if dir == model.TrafficDirectionInboundVIP {
+		inScope := waypointInScope(b.proxy, e)
+		if inScope {
```
Comment (stevenctl) [inline pilot/pkg/xds/endpoint_builder.go:463]: Seems flipped

Comment (howardjohn) [reply pilot/pkg/xds/endpoint_builder.go:463]: Good catch, probably why it fails tests :-) thought it was a flake we were seeing (fix in progress)

**Review by stevenctl — APPROVED** [review]

**Pull request merged**